Makefile.core.mk

```diff
@@ -33,12 +33,12 @@ export ISTIO_IMAGE_VERSION
 ISTIO_SHA ?= $(shell < ${ISTIOIO_GO}/go.mod grep 'istio.io/istio v' | cut -d'-' -f3)
```

Comment:
We should be matching this, right? Why are we using the 1.6.0-beta.0 tag? I would expect this PR to do nothing more than update go.mod with the SHA of 1.6.0-beta.0

Reply:
@howardjohn If you know of a way to get from the sha in go.mod (short) to a valid image tag  long SHA, then we can do this on non-releases.

See my issue: https://github.com/istio/istio.io/issues/7081

If one unsets HUB and TAG one would hope things work, but the calculated image name does not, so all you get is a failure.  The CI/CD incorrectly passes in `latest` so the images can be found, but the tests are now running images from who knows where.

Reply:
@ericvn two possible options:

1. if we have istio/istio repo cloned, we can resolve short SHA -> long SHA
2. We can just have the full SHA defined in the makefile to match the go.mod. We can add a simple assertion they are kept in sync

Reply:
@howardjohn Yes, I know how to get the long sha from the short. I can add code in the init.sh to do that. The main issue is that certain skas will have images from one image repository (those like 1.6-alpha.1d3b136325a5592511f72a9b041ea396bdbd5180) while some will have images from another repository (like 1.6.0-beta.0). I'm not sure that I can figure out which from the sha alone, other that to see if I can pull the image.

Reply:
Every single SHA will have `gcr.io/istio-testing`. We shouldn't be using 1.6.0-beta.0, we should be using `f7c106d20f70a0f7ef9720c790cc5c20230c33fd`, which will exist as 1.6-alpha.f7c106d20f70a0f7ef9720c790cc5c20230c33fd. There is no need to special case beta/alpha releases

Reply:
OK. I looked and there is indeed a set of `1.6-alpha.f7c106d20f70a0f7ef9720c790cc5c20230c33fd` images which should match the `1.6.0-beta.0` images from the other repo. I'll make a change to calculate the longer SHA using the cloned repo and then we can remove that code. And it fixes my other issue.

Reply:
Drafting while I make those changes.

Reply:
OK. If run locally without setting HUB or TAG, the make will use the daily build repo and calculates the long SHA from the short SHA in the go.mod file and will use that just fine.

For some reason, the values are not getting set when called via the prow script in the prow builds, so they are still hardcoded there.

Reply:
Actually, the `eval` is getting evaluated too early when using prow. Things work fine for me locally calling `prow/integ-suite-kind.sh` so not exactly sure what the problem is. More investigation coming. May have to find a makefile and prow expert.

Reply:
Thankfully, you have both a Makefile expert and Prow expert in this review 👍
Insert <tab>`env` to display the environment between these. two lines: https://github.com/istio/istio.io/blob/master/tests/tests.mk#L37-L38.

Run through prow and capture the output.

Then please try adding a `-` preceeding this line: https://github.com/istio/istio.io/blob/master/Makefile.core.mk#L150

Run through prow and capture the output.

Diff the `env` output. This may yield the answer you are looking for.

Please note, there is a way to do this without prow which is a little difficult to use, but easier to capture: https://github.com/istio/istio/wiki/Working-with-Prow#testing-changes-locally I am not sure if this method is the same as the integration test you are running locally. The linked method mimics Istio's CI operation.
Cheers
-steve

Reply:
I will try that shortly. What I notice in the prow logs is the second line here:
```
+ make test.kube.presubmit
/bin/bash: line 0: cd: /home/prow/go/src/istio.io/istio: No such file or directory
ISTIO_SHA = f7c106d20f70
HUB = gcr.io/istio-testing
ISTIOIO_GO=/home/prow/go/src/istio.io/istio.io
ISTIO_GO=/home/prow/go/src/istio.io/istio
```
I don't see this when running locally. I believe that's the eval line failing when it's run early (before the init.sh has created the directory and cloned the repo there).

Local runs (where I see the cp: preserving times instead).
```
+ make test.kube.presubmit
cp: preserving times for '/home/.kube': Read-only file system
ISTIO_SHA = f7c106d20f70
HUB = gcr.io/istio-testing
ISTIOIO_GO=/work
```

Reply:
Thanks to @howardjohn and @sdake I was able to come up with a way to calculate the long SHA from the short sha in the go.mod file and use it within the Makefile. This allows the updating of the base framework from a new istio/istio commit by `go get istio.io/istio@<commit|laber>` followed by a `go mod tidy`. Local tests and prow will use the commit as part of the image name, so no additional changes should be required.